test/auth.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import { auth as native, http as native_http } from '../lib/index';
+import { AwsCredentialsProvider } from '../lib/native/auth';
+
+test('AwsSigningConfig properties', () => {
+    const now = new Date();
+    const provider = AwsCredentialsProvider.newStatic("key", "secret");
+    let config = new native.AwsSigningConfig(
+        native.AwsSigningAlgorithm.SigV4QueryParam,
+        provider,
+        'us-east-1',
+        'iotcore',
+        now,
+        ["abc"],
+        true,
+        true,
+        true,
+    );
```

Comment:
I thought you were going to do something like:
```
let config = new native.AwsSigningConfig({
    algorithm: native.AwsSigningAlgorithm,
    provider: provider,
    region: 'us-east-1',
    ...
});
```
so the 9 args were named and users could leave them out to get defaults?

Reply:
Will revisit once Bret finishes unfactoring signing.